### DISCUSSION TITLE
Trimming Support

### DISCUSSION THREAD

**Discussion by ds5678** [SixLabors/ImageSharp#2162] [Ideas]
[Trimming](https://docs.microsoft.com/en-us/dotnet/core/deploying/trimming/prepare-libraries-for-trimming) allows the compiler to remove unused code and produce smaller applications. Although ImageSharp does not currently support that, adding support is almost trivial since ImageSharp uses very little reflection.

Pull Request: #2160

**Comment by JimBobSquarePants** [comment]
I'm interested in this actually as adding the attributes enables very useful analyzers. 

```xml
<EnableTrimAnalyzer>true</EnableTrimAnalyzer>
```

Maybe we'll one day see an end to the issues we've had with AOT
